typing/subst.mli

```diff
@@ -80,10 +93,33 @@ val modtype_declaration:
   scoping -> t -> modtype_declaration -> modtype_declaration
 val module_declaration: scoping -> t -> module_declaration -> module_declaration
 
-(* Composition of substitutions:
-     apply (compose s1 s2) x = apply s2 (apply s1 x) *)
+(** Composition of substitutions:
+     apply (compose s1 s2) x = apply s2 (apply s1 x) **)
 val compose: t -> t -> t
 
+module Local: sig
+
+  (** Only local substitution may expand an identifier to a generic module
+      type. *)
+  val add_modtype: Ident.t -> module_type -> 'any s -> local
+  val add_modtype_path: Path.t -> module_type -> 'any s -> local
+
+  type error =
+    | Fcm_type_substituted_away of Path.t
+
+  (** Local substituin may fail to expand a module type path *)
+  type 'a res := ('a, error) result
+
+  val signature_item: scoping -> local -> signature_item -> signature_item res
+
+  val signature: scoping -> local -> signature -> signature res
+
+  val compose: local -> local -> local res
+  (** Due to the eager nature of composition for module types, composition of
+      local substitution may fail. *)
```

Comment:
The way I read this is "this may be eager, and so it may fail". But as a user, I may want the *guarantee* that this is eager, so that it fails in my code when I expect it to fail. (The alternative in some cases is not to fail later, but to never fail because some ill-formed types appear during composition but not in the final result.)